sszgen/tag_parser.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"text/scanner"
+)
+
+type tokenState int
+
+const (
+	tsBegin tokenState = iota
+	tsLabel
+	tsValue
+	tsCloseTick
+)
+
+type TagParser struct {
+	sc scanner.Scanner
+	buffer string
+}
+
+func (tp *TagParser) Init(tag string) {
+	sr := strings.NewReader(tag)
+	tp.sc = scanner.Scanner{}
```

Comment:
Do we need to handle sc.Err() anywhere?

Reply:
haha, good idea :) I think the simplest change is to add `tp.sc.Err()` to the return of `GetSSZTags`. I'll make that change now.

Reply:
actually refactored this whole TagParser type out of existence. I can't remember why I wrote it this way, with the weird Init and all, so I just inlined it all into the `GetSSZTags`, which now early returns any errors that the scanner encounters.